### PR TITLE
Add support for pre and post linear modules in `create_mlp`

### LINKED ISSUE #1069: [Question] Policies with layer normalization
### Question

Will there be there be future plans to implement layer normalized policies (MlpLnLstmPolicy, CnnLnLstmPolicy)?

### Checklist

- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by araffin**
## Description

Related to #1069 and https://github.com/DLR-RM/stable-baselines3/pull/1036 and https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/243 (DroQ and CrossQ)

I also added missing tests for `create_mlp`.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #1069 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/torch_layers.py [hunk 1]
```diff
@@ -128,20 +131,48 @@ def create_mlp(
...
+    pre_linear_modules = pre_linear_modules or []
+    post_linear_modules = post_linear_modules or []
+
+    modules = []
     if len(net_arch) > 0:
-        modules = [nn.Linear(input_dim, net_arch[0], bias=with_bias), activation_fn()]
-    else:
-        modules = []
+        for module in pre_linear_modules:
+            modules.append(module(input_dim))
```
Comment (qgallouedec) [inline stable_baselines3/common/torch_layers.py:151]: Is the input dim the same for all modules?

Comment (qgallouedec) [reply stable_baselines3/common/torch_layers.py:151]: I think I get it. It only allows modules that have the same input/output dimension.

**Review by qgallouedec — APPROVED** [review]

File: stable_baselines3/common/torch_layers.py (hunk 1 above)
Comment (araffin) [reply stable_baselines3/common/torch_layers.py:151]: maybe it is clearer with this test: https://github.com/DLR-RM/stable-baselines3/blob/3b84f71bb2b47ca2dc657c7c5205901b6c061f30/tests/test_custom_policy.py#L102-L104

shall I add a comment to avoid confusion?

File: stable_baselines3/common/torch_layers.py [hunk 2]
```diff
@@ -128,20 +131,48 @@ def create_mlp(
...
     :param with_bias: If set to False, the layers will not learn an additive bias
-    :return:
+    :param pre_linear_modules: List of nn.Module to add before the linear layers,
+        for instance, BatchNorm layers.
+        Compared to post_linear_modules, they are used before the output layer (output_dim > 0).
+        The number of input features is passed to the module's constructor.
+    :param post_linear_modules: List of nn.Module to add after the linear layers (and before the activation function),
+        for instance, Dropout or LayerNorm layers.
+        They are not used after the output layer (output_dim > 0).
+        The number of input features is passed to the module's constructor.
```
Comment (qgallouedec) [inline stable_baselines3/common/torch_layers.py:141]:
```suggestion
    :param pre_linear_modules: List of nn.Module to add before the linear layers.
        These modules should maintain the input tensor dimension (e.g. BatchNorm).
        The number of input features is passed to the module's constructor.
        Compared to post_linear_modules, they are used before the output layer (output_dim > 0).
    :param post_linear_modules: List of nn.Module to add after the linear layers 
        (and before the activation function). These modules should maintain the input 
        tensor dimension (e.g. Dropout, LayerNorm). They are not used after the 
        output layer (output_dim > 0). The number of input features is passed to 
        the module's constructor.
```

File: stable_baselines3/common/torch_layers.py (hunk 1 above)
Comment (qgallouedec) [reply stable_baselines3/common/torch_layers.py:151]: Yes! I've also suggested a new documentation: https://github.com/DLR-RM/stable-baselines3/pull/1975/files#r1686213647
